src/installer/src/main.rs

```diff
@@ -0,0 +1,232 @@
+use std::env;
+use std::fs;
+use std::io::{Error, ErrorKind};
+use std::path::{Path, PathBuf};
+use walkdir::WalkDir;
+
+#[derive(Serialize, Deserialize, Debug)]
+struct GhidraConfig {
+    ghidra_path: PathBuf,
+}
+
+/// Copies src/config.json to specified location
+fn copy_config_json(location: &Path) -> Result<()> {
+    let repo_dir = env::current_dir().unwrap();
+    std::fs::copy(
+        &repo_dir.join("src/config.json"),
+        location.join("config.json"),
+    )?;
+    Ok(())
+}
+
+/// Returns vector of os-specific locations
+fn get_search_locations() -> Vec<PathBuf> {
+    let mut locations: Vec<PathBuf> = Vec::new();
+    let base_dirs = BaseDirs::new().unwrap();
+    let user_dirs = UserDirs::new().unwrap();
+
+    locations.push(base_dirs.data_dir().to_path_buf());
+    locations.push(base_dirs.data_local_dir().to_path_buf());
+    locations.push(base_dirs.home_dir().to_path_buf());
+
+    if let Some(path) = base_dirs.executable_dir() {
+        locations.push(path.to_path_buf());
+    }
+    if let Some(path) = user_dirs.desktop_dir() {
+        locations.push(path.to_path_buf());
+    }
+    if let Some(path) = user_dirs.download_dir() {
+        locations.push(path.to_path_buf());
+    }
+    if let Some(path) = user_dirs.document_dir() {
+        locations.push(path.to_path_buf());
+    }
+    if let Some(path) = user_dirs.public_dir() {
+        locations.push(path.to_path_buf());
+    }
+
+    let candidate = Path::new("/opt");
+    if candidate.exists() {
+        locations.push(candidate.to_path_buf());
+    }
+    let candidate = Path::new("/usr");
+    if candidate.exists() {
+        locations.push(candidate.to_path_buf());
+    }
+
+    locations
+}
+
+/// Returns None if Ghidra were not found. Else returns path to Ghidra, which might be user selected.
+fn find_ghidra() -> Result<PathBuf> {
+    let mut ghidra_locations: Vec<PathBuf> = get_search_locations()
+        .into_iter()
+        .filter_map(|x| search_for_ghidrarun(&x))
+        .collect();
+
+    match ghidra_locations.len() {
+        0 => Err(Error::new(ErrorKind::NotFound, "Ghidra not found").into()),
+        1 => Ok(ghidra_locations.pop().unwrap()),
+        _ => select_ghidra_version(ghidra_locations),
+    }
+}
+
+/// Searches for a file containing "ghidraRun" at provided path recursively.
+fn search_for_ghidrarun(entry_path: &Path) -> Option<PathBuf> {
+    for entry in WalkDir::new(entry_path)
+        .into_iter()
+        .filter_map(|e| e.ok())
+        .filter(|e| e.metadata().unwrap().is_file())
+    {
+        if entry.file_name().to_str().unwrap().contains("ghidraRun") {
+            let mut hit = entry.into_path();
+            hit.pop();
+            return Some(hit);
+        }
+    }
+    None
+}
+
+/// Determines Ghidra versions and provides selection interface for the user.
+fn select_ghidra_version(mut ghidra_locations: Vec<PathBuf>) -> Result<PathBuf> {
+    let mut i = 0;
+    while i < ghidra_locations.len() {
```

Comment:
This should be a for-loop instead of a while-loop with manually incremented loop counter.